src/transformers/training_args.py

```diff
@@ -477,7 +490,13 @@ def _setup_devices(self) -> Tuple["torch.device", int]:
         else:
             # Here, we'll use torch.distributed.
             # Initializes the distributed backend which will take care of synchronizing nodes/GPUs
-            torch.distributed.init_process_group(backend="nccl")
+            #
+            # deepspeed performs its own DDP internally, and requires the program to be started with:
+            # deepspeed  ./program.py
+            # rather than:
+            # python -m torch.distributed.launch --nproc_per_node=2 ./program.py
+            if not self.deepspeed:
+                torch.distributed.init_process_group(backend="nccl")
```

Comment:
Can't make a suggestion that spans over non-modified code but I think this would be clear with
```
elif not self.deepspeed:
    # Comment goes here
    torch.distributed.init_process_group(backend="nccl")
```

Reply:
We can. I just find that flattening nested else-ifs often takes away from clear readability. I coded it this way so that it's clear that this branch of else is all the distributed things and then nuancing inside of it with its own if/else, but if you feel the approach you suggest is better or that's the overall style of transformers, then I'll change it to what you suggest - it's a minor thing.

Reply:
I have no strong opinion either, so if you prefer it that way, you're the boss ;-)